packages/html/src/matcher.ts

```diff
@@ -0,0 +1,44 @@
+import isEqual from 'lodash.isequal';
+
+export type Matcher<T> = NewMatcher<T> | OldMatcher<T>;
+
+export interface NewMatcher<T> {
+  match(actual: T): boolean;
+  description(): string;
+  format?(): never;
+}
+
+export interface OldMatcher<T> {
+  match(actual: T): boolean;
+  format(): string;
+  description?(): never;
+}
+
+export type MaybeMatcher<T> = Matcher<T> | T;
+
+export function isMatcher<T>(value: MaybeMatcher<T>): value is NewMatcher<T> {
+  return value && typeof (value as NewMatcher<T>).match === 'function' && typeof (value as NewMatcher<T>).description === 'function';
+}
+
+export function isDeprecatedMatcher<T>(value: MaybeMatcher<T>): value is OldMatcher<T> {
```

Comment:
We need to record the work to remove deprecated marchers and also the applyMatcher api

Reply:
https://github.com/thefrontside/interactors/issues/5